### PR TITLE
Try to work around Visual Studio unary unsigned minus bug

### PR DESCRIPTION
**Pull request by rcurtin**
This is an attempt to work around the issue reported in #3356.

### CODE REVIEW

**Comment by Blakjak88** [comment]
I tested this change on Visual Studio 2022 and it removes the compilation error mentioned in #3356

**Comment by rcurtin** [comment]
Awesome!  Thanks for testing.

File: src/mlpack/core/data/load_numeric_csv.hpp [hunk 1]
```diff
@@ -14,8 +14,32 @@
...
 
-namespace mlpack{
-namespace data{
+namespace mlpack {
+namespace data {
+
+/**
+ * A safe function to get negative or positive infinity, which avoids unary
+ * minus on an unsigned type.  This works around a Visual Studio bug.
+ * (TODO: add a link?)
```
Comment (zoq) [inline src/mlpack/core/data/load_numeric_csv.hpp:23]: Is this something you want to do before merging this?

Comment (rcurtin) [reply src/mlpack/core/data/load_numeric_csv.hpp]: Thanks for pointing that out... I had forgotten to handle that.  After digging around a bit more, it's not an error, it's just a warning, and I don't have a relevant link for it.  I fixed my comment in 45394e01c6. :+1:

**Review by zoq — APPROVED** [review]
Looks good to me.

**Review by shrit — APPROVED** [review]

**Pull request merged**